src/App.js

```diff
@@ -788,18 +789,39 @@ export default class App extends Component {
     const response = await fetch(endpoint);
     const body = await response.json();
     const {
-      proposals: proposalsList,
+      proposals,
       voteEndTime,
-      voteStartTime
+      voteStartTime,
+      trashAddress
     } = body.contents;
+
+    const proposalsList = proposals.map((proposal, i)=>{
+      return {
+        ...proposal,
+        _id: i
+      }
+    });
+
+    console.log({proposalsList})
+
+    const proposalsDictionary = proposalsList.reduce((acc,item)=>{
+      const {_id} = item;
+      acc[_id] = item;
+      return acc;
+    },{});
+
+    console.log({proposalsDictionary});
+
     this.setState(state => ({
       ...state,
```

Comment:
I've already fixed this in #24, but you use it wrong. You don't need to pass final state to setState — only what you want to change. Also, you don't need to use callback form of setState here — it only needed when your update depends on current state (increment, toggle and so on)